Add a get IRC27NFT Metadata guide (#1583)
iotaledger/iota-wiki: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md

Reviewer: This file should get the .mdx extension as it used mdx stuff.

---
description: How to get NFT metadata on L1
image: /img/logo/WASP_logo_dark.png
tags:
  - NFT
  - EVM
  - how-to
---
import GetNftMetadata from '../../../_partials/how-tos/token/_get-nft-metadata.md';

# Get NFT Metadata

> 

Reviewer: ```suggestion

:::info ERC721

As your L1 NFT is always registered as [ERC721](https://eips.ethereum.org/EIPS/eip-721), you might want to get the metadata like `tokenURI` from there. Using `getIRC27NFTData` is normally only needed if you need special [IRC27](https://wiki.iota.org/tips/tips/TIP-0027/) metadata

```

Author: Done, thanks

---

> description: How to get NFT metadata on L1

Reviewer: ```suggestion
description: How to get NFT metadata from a L1 NFT
```
```suggestion
description: How to get NFT metadata on L1
```

image: /img/logo/WASP_logo_dark.png
tags:
  - NFT
  - EVM
  - how-to
---
import GetNftMetadata from '../../../_partials/how-tos/token/_get-nft-metadata.md';
import ERC721Admonition from '../../../_admonitions/_ERC721.md';

> # Get NFT Metadata

Reviewer: Should we rename this to:
```suggestion
# Get IRC27 NFT Metadata
```
Would make it more clear that we are not talking about standard ERC721 or similar here

<ERC721Admonition />

This guide explains how to use the [`getIRC27NFTData`](../../../reference/magic-contract/ISCSandbox.md#getirc27nftdata) function within a smart contract to fetch information about a specific IRC27 NFT on the IOTA Network.

<GetNftMetadata />

## Understanding the `getIRC27NFTData` Function

The [`getIRC27NFTData`](../../../reference/magic-contract/ISCSandbox.md#getirc27nftdata) function retrieves metadata for an IRC27 NFT based on its identifier. IRC27 is a series of standards to support interoperable and universal NFT systems throughout the IOTA ecosystem.

## How To Use `getIRC27NFTData`

> ### 1. Fetch IRC27 NFT Metadata
> 

Reviewer: ```suggestion
```

Not needed if we only have one title anymore

> Create a function called `fetchNFTData` in your contract that calls `getIRC27NFTData` and processes its return value.

Reviewer: ```suggestion
Create a function called `fetchNFTData` in your contract that calls `getIRC27NFTData` and processes its return value. getIRC27NFTData returns a struct of type [`IRC27NFTMetadata`](../../../reference/magic-contract/ISCTypes.md#irc27nftmetadata) which contains properties like the NFT name, uri and more

:::info IRC27NFTMetadata URI

The uri property contains a JSON object which follows the `ERC721` standard. This JSON is also returned by the [`tokenURI`](../../../reference/magic-contract/ERC721NFTs.md#tokenuri) function from the `ERC721NFTs` contract.

:::
```

```solidity
function fetchNFTData(bytes memory nftID) public view returns (IRC27NFT memory irc27NftData) {
    bytes32 id = bytes32(nftID);
    NFTID nftIDTyped = NFTID.wrap(id);
    irc27NftData = ISC.sandbox.getIRC27NFTData(nftIDTyped);
    return irc27NftData;
}
```

> ### 2. Encode the URI Data in `IRC27NFTMetadata`

Reviewer: As agreed on Slack, let's remove the whole on-chain encode and decoding stuff
